src/installLogsPrinter.js

```diff
@@ -72,15 +76,25 @@ function installLogsPrinter(on, options = {}) {
         allMessages[data.spec][data.test] = messages;
       }
 
-      logToTerminal(messages, options);
+      if (options.printLogsToConsole === "always" ||
+          ((options.printLogsToConsole === "onFail" || typeof options.printLogsToConsole === 'undefined')
+              && state !== "passed")){
+        logToTerminal(messages, options);
+      }
+
       return null;
     },
     [CONSTANTS.TASK_NAME_OUTPUT]: () => {
       outputProcessors.forEach((processor) => {
-        processor.write(allMessages);
-        logOutputTarget(processor);
+        if (options.printLogsToFile === "always" ||
+            ((options.printLogsToFile === "onFail" || typeof options.printLogsToFile === 'undefined')
```

Comment:
Instead of checking here for undefined please set the default value eariler to onFail.